components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -719,6 +719,28 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
         return callbackURL;
     }
 
+    public static String getAccessUrl(org.wso2.carbon.identity.recovery.model.Property[] properties)
+            throws UnsupportedEncodingException, URISyntaxException {
+
+        if (properties == null) {
+            return null;
+        }
+        String accessURL = null;
+        for (org.wso2.carbon.identity.recovery.model.Property property : properties) {
+            if (IdentityRecoveryConstants.ACCESS_URL.equals(property.getKey())) {
+                accessURL = property.getValue();
+                break;
+            }
+        }
+
+        if (StringUtils.isNotBlank(accessURL)) {
+            URI uri = new URI(accessURL);
+            accessURL = new URI(uri.getScheme(), uri.getAuthority(), uri.getPath(), null, null)
+                    .toString();
+        }
+        return accessURL;
+    }
```

Comment:
shall we use Optional here?

Reply:
Changed accessUrl property to isAccessUrlAvailable boolean property by https://github.com/wso2-extensions/identity-governance/pull/787/commits/47e973bf1bd8817ab28abe25549992ec43e49d2b. Hence no need to use optional there